numpy/core/src/umath/matmul.c.src

```diff
@@ -79,24 +81,30 @@ NPY_NO_EXPORT void
     enum CBLAS_ORDER order = CblasRowMajor;
     enum CBLAS_TRANSPOSE trans1, trans2;
     int M, N, P, lda, ldb;
-    M = m;
-    N = n;
-    P = p;
+    assert(m < NPY_MAX_INT);
+    assert(n < NPY_MAX_INT);
+    assert(p < NPY_MAX_INT);
```

Comment:
Maybe these should be `<=`, but I suppose there's no harm in being conservative here, and who knows whether blas behaves at this end of the range.